cedar-drt/fuzz/src/lib.rs

```diff
@@ -43,6 +43,64 @@ use std::collections::HashSet;
 pub const RUST_AUTH_MSG: &str = "rust_auth (ns) : ";
 pub const RUST_VALIDATION_MSG: &str = "rust_validation (ns) : ";
 
+/// Compare the behavior of the partial evaluator in `cedar-policy` against a custom Cedar
+/// implementation. Panics if the two do not agree. `expr` is the expression to
+/// evaluate and `request` and `entities` are used to populate the evaluator.
+pub fn run_pe_test(
+    custom_impl: &impl CedarTestImplementation,
+    request: ast::Request,
+    expr: &ast::Expr,
+    entities: &Entities,
+    enable_extensions: bool,
+) {
+    let exts = if enable_extensions {
+        Extensions::all_available()
+    } else {
+        Extensions::none()
+    };
+
+    let eval = Evaluator::new(request.clone(), entities, &exts);
+    use cedar_policy_core::ast::PartialValue;
+    use cedar_testing::cedar_test_impl::ExprOrValue;
+    use log::debug;
+    let expected = match eval.partial_interpret(expr, &std::collections::HashMap::default()) {
+        Ok(PartialValue::Value(v)) => Some(ExprOrValue::value(v)),
+        Ok(PartialValue::Residual(r)) => Some(ExprOrValue::Expr(r)),
+        Err(_) => None,
+    };
+    debug!("Expected: {:?}", expected);
+
+    let definitional_res = custom_impl.partial_evaluate(
+        &request,
+        entities,
+        expr,
+        enable_extensions,
+        expected.clone(),
+    );
+    match definitional_res {
+        TestResult::Failure(err) => {
+            // TODO(#175): Ignore cases where the definitional code returned an error due to
+            // an unknown extension function.
+            //      if err.contains("jsonToExtFun: unknown extension function") {
+            //         return;
+            //    }
+            // No other errors are expected
```

Comment:
I'm confused whether this is still a TODO or not, since the relevant code seems to be commented out

Reply:
Mistake, commented out for debugging.